randovania/games/super_metroid/patcher/super_duper_metroid_patcher.py

```diff
@@ -151,15 +185,44 @@ def create_patch_data(self, description: LayoutDescription, players_config: Play
             visual_etm=pickup_creator.create_visual_etm(),
         )
 
+        specific_patch_list = [attr for attr in vars(configuration.patches) if
```

Comment:
This function really needs some tests. At the very least, have a rdvgame file for SM in the test files dir, the expected output json and a test that does the stuff for that rdvgame.